src/current/v24.3/triggers.md

```diff
@@ -0,0 +1,505 @@
+When multiple triggers activate on the same table, they behave as follows:
+
+- The triggers activate in alphabetical order by trigger name.
+- The output of a `BEFORE` trigger is passed to the next `BEFORE` trigger. For details on values returned by triggers, refer to [Trigger function](#trigger-function).
+
+For an example, refer to [Demonstrate `BEFORE` and `AFTER` trigger ordering](#demonstrate-before-and-after-trigger-ordering).
+
+### Trigger function
+
+A trigger executes a [function]({% link {{ page.version.version }}/user-defined-functions.md %}) called a trigger function. A trigger function is defined with [`CREATE FUNCTION`]({% link {{ page.version.version }}/create-function.md %}#create-a-trigger-function) and has the following requirements:
+
+- The function must return type `TRIGGER`. 
+- The function must be declared without arguments.
+- The function must be written in [PL/pgSQL]({% link {{ page.version.version }}/plpgsql.md %}).
+- The function for a [`BEFORE`](#trigger-conditions) trigger must return one of the following values:
+	- The [`NEW`](#trigger-variables) table row resulting from the SQL operation that activated the trigger. This variable applies only to `INSERT` and `UPDATE` triggers, and also allows the `BEFORE` trigger to modify the row before it is written.
+	- The [`OLD`](#trigger-variables) table row affected by the SQL operation that activated the trigger. This variable applies only to `UPDATE` and `DELETE` triggers.
+	- `NULL`, which stops the SQL operation that activated the `BEFORE` trigger.
+- The function for an [`AFTER`](#trigger-conditions) trigger should return `NULL`, because its return value will be ignored.
+- The function must be defined before creating the trigger.
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE OR REPLACE FUNCTION function_name()
+  RETURNS TRIGGER AS $$
+  BEGIN
+	...
+  END
+  $$ LANGUAGE PLpgSQL;
+~~~
+
+Refer to [Examples](#examples).
+
+#### Trigger variables
+
+The following trigger variables are automatically created for trigger functions, and can be used in the function body.
+
+|      Variable     |    Type    |                                                                                      Description                                                                                      |
+|-------------------|------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `NEW`             | `RECORD`   | New table row resulting from the SQL operation. For `INSERT` triggers, this is the row that will be inserted. For `UPDATE` triggers, this is the row containing the updated values.   |
+| `OLD`             | `RECORD`   | Old table row affected by `UPDATE` and `DELETE` operations. For `UPDATE` triggers, this is the row that will be updated. For `DELETE` triggers, this is the row that will be deleted. |
```

Comment:
Unless it's mentioned elsewhere, we should mention that `NEW` is NULL for DELETE triggers, and `OLD` is NULL for INSERT triggers.